File: tests/src/main/java/com/github/klee0kai/test/mowgli/MoonSky.java
package com.github.klee0kai.test.mowgli;

import com.github.klee0kai.test.mowgli.galaxy.*;

import javax.inject.Inject;

public class MoonSky {

    @Inject
    public IStar star;

    @Inject
    public Sun sun;


    @Inject
    public IPlanet planet;

    @Inject
    public Mercury mercury;

    @Inject
    public Earth earth;


}

File: tests/src/main/java/com/github/klee0kai/test/mowgli/MoonSky.kt
package com.github.klee0kai.test.mowgli

import com.github.klee0kai.test.mowgli.galaxy.*
import javax.inject.Inject

class MoonSky {
    @JvmField
    @Inject
    var star: IStar? = null

    @JvmField
    @Inject
    var sun: Sun? = null


    @JvmField
    @Inject
    var planet: IPlanet? = null

    @JvmField
    @Inject
    var mercury: Mercury? = null

    @JvmField
    @Inject
    var earth: Earth? = null
}
